core/templates/dev/head/pages/state_editor/StateHintsEditorDirective.js

```diff
@@ -30,21 +30,24 @@ oppia.directive('stateHintsEditor', [
       controller: [
         '$scope', '$rootScope', '$uibModal', '$filter',
         'GenerateContentIdService', 'AlertsService', 'INTERACTION_SPECS',
-        'StateHintsService', 'COMPONENT_NAME_HINT',
+        'StateHintsService', 'COMPONENT_NAME_HINT', 'StateEditorService',
         'StateContentIdsToAudioTranslationsService', 'EditabilityService',
         'StateInteractionIdService', 'UrlInterpolationService',
         'HintObjectFactory', 'StateSolutionService',
         function(
             $scope, $rootScope, $uibModal, $filter,
             GenerateContentIdService, AlertsService, INTERACTION_SPECS,
-            StateHintsService, COMPONENT_NAME_HINT,
+            StateHintsService, COMPONENT_NAME_HINT, StateEditorService,
             StateContentIdsToAudioTranslationsService, EditabilityService,
             StateInteractionIdService, UrlInterpolationService,
             HintObjectFactory, StateSolutionService) {
           $scope.EditabilityService = EditabilityService;
           $scope.StateHintsService = StateHintsService;
           $scope.activeHintIndex = null;
           $scope.canEdit = GLOBALS.can_edit;
+          if (StateEditorService.getInQuestionMode()) {
```

Comment:
What if someone without edit rights goes to the question editor page?

Reply:
Done, made this EditabilityServices.isEditable().